src/main/java/io/github/oliviercailloux/y2018/jbiblio/j_biblio/services/ItemJsonServlet.java

```diff
@@ -0,0 +1,99 @@
+package io.github.oliviercailloux.y2018.jbiblio.j_biblio.services;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import io.github.oliviercailloux.y2018.jbiblio.j_biblio.basicentities.Item;
+
+@SuppressWarnings("serial")
+@WebServlet("item")
+public class ItemJsonServlet extends HttpServlet {
+	private static final Logger LOGGER = Logger.getLogger(ItemJsonServlet.class.getCanonicalName());
+
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+		resp.setCharacterEncoding(StandardCharsets.UTF_8.name());
+		resp.setContentType(MediaType.APPLICATION_JSON);
+		resp.setLocale(Locale.ENGLISH);
+		/**
+		 * Converting to Json format
+		 */
+		try (Jsonb jsonb = JsonbBuilder.create();) {
+
+			/**
+			 * Jpa will be implemented in the next sprint
+			 */
+			Item item = this.initItem(req);
+			String jsonItem = jsonb.toJson(item);
+			resp.setStatus(HttpServletResponse.SC_OK);
+			resp.getWriter().println(jsonItem);
+			LOGGER.info(" Display object in JSON format " + jsonItem);
+
+		} catch (Exception e) {
+			// Send error status
+
+			LOGGER.warning(" Error  " + e.toString());
+			resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+			resp.sendError(HttpServletResponse.SC_BAD_REQUEST, "The research failed");
+
+		}
+	}
+
+	@Override
+	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		resp.setCharacterEncoding(StandardCharsets.UTF_8.name());
+		resp.setContentType(MediaType.TEXT_PLAIN);
+		resp.setLocale(Locale.ENGLISH);
+		try (Jsonb jsonb = JsonbBuilder.create();) {
+			try (BufferedReader reader = req.getReader()) {
+				/**
+				 * Jpa will be implemented in the next sprint
+				 */
+				@SuppressWarnings("unused")
+				Item item = jsonb.fromJson(reader, Item.class);
+			}
+			resp.setStatus(HttpServletResponse.SC_OK);
+			resp.getWriter().println("The object is successfully inserted");
```

Comment:
Bien, mais faites plutôt savoir à l’appelant que l’insertion n’est pas encore implémentée, pour éviter les confusions.

Reply:
Modification en "The object is successfully initialized. Database insertion  will be implemented in the next sprint"